src/main/java/edu/kcc/animal/Animal.java

```
@@ -125,7 +125,9 @@ public void setFixed(boolean fixed) {
     
     // TODO: Need fixedValidator method - don't allow an animal that is already fixed to be fixed again 
     private void fixedValidator(boolean fixed) {
-        
+        if(fixed == true){
```

The code has two "fixed" variables: (1) the instance variable, and (2) the parameter. You only want to throw the exception when the instance variable is true and the parameter is false -- this would mean that we are trying to un-neuter an animal.